components/result/result.component.ts

```diff
@@ -46,42 +45,51 @@ const ExceptionStatus = ['404', '500', '403'];
   exportAs: 'nzResult',
   template: `
     <div class="ant-result-icon">
-      <ng-container *ngIf="!isException; else exceptionTpl">
-        <ng-container *ngIf="icon">
+      @if (!isException) {
+        @if (icon) {
           <ng-container *nzStringTemplateOutlet="icon; let icon">
             <span nz-icon [nzType]="icon" nzTheme="fill"></span>
           </ng-container>
-        </ng-container>
-        <ng-content *ngIf="!icon" select="[nz-result-icon]"></ng-content>
-      </ng-container>
+        } @else {
+          <ng-content select="[nz-result-icon]"></ng-content>
+        }
+      } @else {
+        @switch (nzStatus) {
+          @case (404) {
+            <nz-result-not-found />
+          }
+          @case (500) {
+            <nz-result-server-error />
+          }
+          @case (403) {
+            <nz-result-unauthorized />
+          }
+        }
```

Comment:
```suggestion
        @switch (nzStatus) {
          @case ('404') {
            <nz-result-not-found />
          }
          @case ('500') {
            <nz-result-server-error />
          }
          @case ('403') {
            <nz-result-unauthorized />
          }
        }
```

Reply:
indeed, nzStatus is a string thanks for the review, fixed now :)